### PR TITLE
Update to upstream 1.13.0

### PR DESCRIPTION
**Pull request by andreittr**
### Description of changes

Update Google Test to upstream 1.13.0, as well as clean up build scripts & documentation.

### CODE REVIEW

**Review by mariasfiraiala — CHANGES_REQUESTED** [review]
The changes look good, thanks @andreittr. Regarding the documentation part, however, the README specifies the order of the libraries  which we don't use anymore (and has `newlib` mentioned as well - kinda obsolete).

AFAIK, the latest correct order is

```Makefile
LIBS := $(UK_LIBS)/lib-libcxxabi:$(UK_LIBS)/lib-libcxx:$(UK_LIBS)/lib-compiler-rt:$(UK_LIBS)/lib-libunwind:$(UK_LIBS)/lib-musl:$(UK_LIBS)/lib-googletest
```

Do you mind changing that too @andreittr? Actually the whole README should be updated to have the md format. I can add a commit to achieve that, if you find it too cumbersome.

**Comment by andreittr** [comment]
> The changes look good, thanks @andreittr. Regarding the documentation part, however, the README specifies the order of the libraries which we don't use anymore (and has `newlib` mentioned as well - kinda obsolete).
> 
> AFAIK, the latest correct order is
> 
> ```makefile
> LIBS := $(UK_LIBS)/lib-libcxxabi:$(UK_LIBS)/lib-libcxx:$(UK_LIBS)/lib-compiler-rt:$(UK_LIBS)/lib-libunwind:$(UK_LIBS)/lib-musl:$(UK_LIBS)/lib-googletest
> ```
> 
> Do you mind changing that too @andreittr? Actually the whole README should be updated to have the md format. I can add a commit to achieve that, if you find it too cumbersome.

Good point @mariasfiraiala , thanks! I'll add a commit updating the README to be roughly in line with ones in more up-to-date libs. I'll also simplify the lib order paragraph to just "put at the end of the lib list".

File: README.md [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
-have any main function in their unit tests and expect googletest to
-provide it.
+Some libraries that use GoogleTest rely on it to also provide a boilerplate main function to run their tests.
+Providing this function by default might however conflict with the application's intended main.
+For this purpose the LIBGOOGLETEST_BUILD_MAIN config option controls whether GoogleTest provides a `main()` function.
 								 
-Please refer to the `README.md` as well as the documentation in the
-`doc/` subdirectory of the main unikraft repository for further
-information.
+Being a C++ library, libcxx is required.
```
Comment (mariasfiraiala) [inline README.md]:
```suggestion
Being a C++ library, `libcxx` is required.
```

File: README.md [hunk 2]
```diff
@@ -1,19 +1,12 @@
...
+This library should come at the end of the dependency list.
```
Comment (mariasfiraiala) [inline README.md]:
```suggestion
`lib-googletest` should come at the end of the dependency list.
```

It isn't clear enough what library should come last.

**Review by mariasfiraiala — CHANGES_REQUESTED** [review]
Thanks @andreittr, 2 last comments from me below. After this, I think we should merge this.

**Review by mariasfiraiala — APPROVED** [review]
All good now, thanks a lot for the effort, @andreittr!

Reviewed-by: Maria Sfiraiala <maria.sfiraiala@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**